plugins/inputs/nfsclient/nfsclient.go

```diff
@@ -0,0 +1,421 @@
+					fields[item] = t
+				}
+				acc.AddFields("nfs_ops", fields, tags)
+			}
+		}
+	} else {
+		if first == "READ" {
+			fields["read_ops"] = nline[0]
+			fields["read_retrans"] = (nline[1] - nline[0])
+			fields["read_bytes"] = (nline[3] + nline[4])
+			fields["read_rtt"] = nline[6]
+			fields["read_exe"] = nline[7]
+			acc.AddFields("nfsstat_read", fields, tags)
+		} else if first == "WRITE" {
+			fields["write_ops"] = nline[0]
+			fields["write_retrans"] = (nline[1] - nline[0])
+			fields["write_bytes"] = (nline[3] + nline[4])
+			fields["write_rtt"] = nline[6]
+			fields["write_exe"] = nline[7]
+			acc.AddFields("nfsstat_write", fields, tags)
+		}
+	}
+
+	return nil
+}
+
+func (n *NFSClient) processText(scanner *bufio.Scanner, acc telegraf.Accumulator) error {
+	var device string
+	var version string
+	var export string
+	var skip bool
+	var nfs3Ops map[string]bool
+	var nfs4Ops map[string]bool
+	nfs3Ops = make(map[string]bool)
+	nfs4Ops = make(map[string]bool)
+
+	if len(n.IncludeOperations) == 0 {
+		for _, Op := range nfs3Fields {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range nfs4Fields {
+			nfs4Ops[Op] = true
+		}
+	} else {
+		for _, Op := range n.IncludeOperations {
+			nfs3Ops[Op] = true
+		}
+		for _, Op := range n.IncludeOperations {
+			nfs4Ops[Op] = true
+		}
+	}
+
+	if len(n.ExcludeOperations) > 0 {
+		for _, Op := range n.ExcludeOperations {
+			if nfs3Ops[Op] {
+				delete(nfs3Ops, Op)
+			}
+			if nfs4Ops[Op] {
+				delete(nfs4Ops, Op)
+			}
+		}
+	}
+
+	for scanner.Scan() {
+		line := strings.Fields(scanner.Text())
+
+		if in(line, "fstype") && (in(line, "nfs") || in(line, "nfs4")) && len(line) > 4 {
+			device = line[4]
+			export = line[1]
+			skip = false
+		} else if (in(line, "(nfs)") || in(line, "(nfs4)")) && len(line) > 5 {
+			version = strings.Split(line[5], "/")[1]
+		}
+
+		if len(n.IncludeMounts) > 0 {
+			skip = true
+			for _, RE := range n.IncludeMounts {
+				matched, _ := regexp.MatchString(RE, device)
+				if matched {
+					skip = false
+					break
+				}
+			}
+		}
+
+		if len(n.ExcludeMounts) > 0 {
+			for _, RE := range n.ExcludeMounts {
+				matched, _ := regexp.MatchString(RE, device)
+				if matched {
+					skip = true
+					break
+				}
+			}
+		}
+
+		if !skip && len(line) > 0 {
+			n.parseStat(device, export, version, line, n.Fullstat, nfs3Ops, nfs4Ops, acc)
+		}
+	}
+	return nil
+}
+
+func getMountStatsPath() string {
+	path := "/proc/self/mountstats"
+	if os.Getenv("MOUNT_PROC") != "" {
+		path = os.Getenv("MOUNT_PROC")
+	}
+	return path
+}
+
+func (n *NFSClient) Gather(acc telegraf.Accumulator) error {
+	var outerr error
+
+	file, err := os.Open(getMountStatsPath())
+	if err != nil {
+		log.Fatal("Failed opening the /proc/self/mountstats file:  ", err)
+		return err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	n.processText(scanner, acc)
+
+	if err := scanner.Err(); err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	return outerr
```

Comment:
Return `nil` here and get rid of outerr!

Reply:
Old code.  Removed.

Reply:
I don't see it's removed!?!?

Reply:
Removed in 5a57b8c8